### ISSUE TITLE
Usage in Node

### ISSUE THREAD

**Issue by genu** [maizzle/docs#17]
The section on using on server side isn't very clear. I wasn't actually able to get it to work and there doesn't seem to be any other examples.

I think it needs to be more elaborate, or at least provide a full example so that someone can actually write up a good guide in the docs on how to set it up.

**Comment by cossssmin** [comment]
Hi there,

The docs do provide a full [example](https://maizzle.com/docs/server-side/#example), and the options object it receives, is also documented. Can you please let me know what was unclear regarding these, so we can improve the docs? :)

**Comment by genu** [comment]
Hi @cossssmin, 

Thanks for the reply. I was trying to run use the example you reference, but It throws errors about the usage of `master.njk` in the template string. I assume that the `master.njk` file is mentioned in other sections of the documentation, but in this particular case its not clear what it is, where its supposed to be, and how the paths are resolved.

You mentioned in the other issue how paths are resolved, I will try to your suggestion out.  Clarity around that I think would help in the docs, since most of the documentation is, I think, geared toward the CLI usage.

**Comment by cossssmin** [comment]
Yes, `master.njk` is a Layout file (as you can see from the Front Matter `layout:` key), Of course, [Layouts](https://maizzle.com/docs/layouts/) are documented. You're right about clarity on how Nunjucks resolves paths, I'll update that page with a note on this 👍

**Comment by genu** [comment]
I spent some time trying to get a node setup working, but failed.

I used the following config:

```js
import * as maizzle from '@maizzle/framework';
import * as tailwindConfig from '../../../../emails/tailwind.config.js';
import * as maizzleConfigProduction from '../../../../emails/config.production.js';

const tmpl = `
    ---
    title: Using Maizzle on the server
    layout: src/layouts/master.njk
    ---

    {% block template %}
    <table>
      <tr>
        <td class="bg-blue hover-bg-blue-dark text-white text-center rounded">
          <a href="https://maizzle.com" class="text-white inline-block text-sm font-semibold py-16 px-24 no-underline">Confirm email address</a>
        </td>
      </tr>
    </table>
    {% endblock %}`;

maizzle
      .render(tmpl, {
        tailwind: {
          config: tailwindConfig,
          css:
            '@tailwind utilities;',
        },
        maizzle: {
          config: maizzleConfigProduction,
        },
      })
      .then(html => {
        console.log(html);
      });
```

The output of the `html` above is: 
```html
--- title: Using Maizzle on the server layout: src/layouts/master.njk ---
<table>
  <tr>
    <td class="bg-blue hover-bg-blue-dark text-white text-center rounded">
      <a
        href="https://maizzle.com"
        class="text-white inline-block text-sm font-semibold py-16 px-24 no-underline"
        >Confirm email address</a
      >
    </td>
  </tr>
</table>
```

It doesn't seem to pickup the front matter, nor is it applying the layout.

I'm assuming, there are path issue? But there are no errors thrown... :/

Btw, the above is implemented in typescript.

Any thoughts/suggestions?

**Comment by cossssmin** [comment]
Thanks for that, found the issue. It's not about Typescript, it's about Nunjucks `{% extends %}` and the fact that I forgot to implement it in the `render()` method 🤦‍♂️

As a quick workaround, if you want to use a layout, simply add that `extends` to your string that you want rendered (remove the `layout` FM key):

```
const tmpl = `
---
title: Using Maizzle on the server
---

{% extends "src/layouts/default.njk" %}

{% block template %}
  // your HTML...
{% endblock %}`;
```

Note that the `"src/layouts/default.njk"` path is relative to where you execute the command from.

I'll work on fixing this in the coming days, I also noticed there's a small issue with the config as well (it doesn't ensure a _full_ config is generated, instead only takes what you give it + front matter keys... so for a full config you'd need to pass in one yourself).

Thanks for bringing this up and helping improve Maizzle! 😊

**Comment by cossssmin** [comment]
Should be now fixed in `v0.4.0` , please update and let me know how it goes. Docs have also been [updated](https://maizzle.com/docs/server-side/).

**Comment by genu** [comment]
Thanks for the update. I took the latest and it looks like its working now.

I'm still working on the implementation, but I'll be doing more testing in the coming days and provide feedback.

Thanks!

**Comment by genu** [comment]
One thing I notice, is that nunjucks template includes can potentially fail more often because when using node, the working directory will generally be different than what nunjucks expects. (This isn't an issue when running via cli)

Any reference to `src/partials/...` in the template would not be found.

I think an option to pass a working direction to the nunjucks configuration would be helpful here. 

Something like this would work, I think, `framework/src/nunjucks/index.js` but the init() would need to take a config param.

```js
const nunjucksEnv = nunjucks.configure(`${config.basePath || process.cwd()}/src/emails/`, { autoescape: false, noCache: true })
```

Any thoughts?

**Comment by cossssmin** [comment]
We could add an optional `basePath` to `render()` opts (wouldn’t want it in the config files, as it would only be required for this). 

However, this would obviously apply to both layouts and any includes.

Will test and see how it goes.
